src/librustc_privacy/lib.rs

```diff
@@ -971,42 +996,7 @@ impl<'a, 'tcx: 'a> TypeVisitor<'tcx> for SearchInterfaceForPrivateItemsVisitor<'
             }
         }
 
-        if let ty::TyProjection(ref proj) = ty.sty {
-            // Avoid calling `visit_trait_ref` below on the trait,
-            // as we have already checked the trait itself above.
-            proj.trait_ref.super_visit_with(self)
```

Comment:
Reminder that `visit_trait_ref` was *never* called for `TyProjection` in this `impl`.